.github/workflows/iroha2-dev-pr.yml

```diff
@@ -144,44 +150,81 @@ jobs:
+        with:
+          install: true
+      - name: Install Python and Poetry
         run: |
-          cargo build -p iroha_client_cli -p kagami -p irohad
-      - name: Setup test Iroha 2 environment on the bare metal
+          yum install -y python${{ env.PYTHON_VERSION }} python${{ env.PYTHON_VERSION }}-devel
+          curl -sSL https://install.python-poetry.org | python3 -
+          echo 'export PATH="${{ env.POETRY_PATH }}:$PATH"' >> /etc/profile
+          source /etc/profile
+      - name: Install Torii API Dependencies
+        working-directory: torii/pytests
         run: |
-          pip3 install -r scripts/requirements.txt --no-input --break-system-packages
-          ./scripts/test_env.py setup
-      - name: Mark binaries as executable
+          ${{ env.POETRY_PATH }} install
+      - name: Build and Tag Docker Images
+        uses: docker/build-push-action@v6
+        if: always()
+        with:
+          context: .
+          load: true
+          file: Dockerfile
+          tags: |
+            hyperledger/iroha:local
+            hyperledger/iroha:dev
+          cache-from: type=gha
+          cache-to: type=gha,mode=max
+      - name: Test Docker Compose Setup
         run: |
-          chmod +x ${{ env.CLIENT_CLI_DIR }}
-      - name: Install torii api dependencies using Poetry
+          docker compose -f ${{ env.DOCKER_COMPOSE_PATH }}/docker-compose.yml up --wait || exit 1
+      - name: Run Torii API Tests
         working-directory: torii/pytests
         run: |
-          poetry install
-      - name: Run torii api tests
-        working-directory: torii/pytests
+          ${{ env.POETRY_PATH }} run pytest
+      - name: Copy Client CLI Binary from Iroha Container
+        run: |
+          mkdir -p ${{ env.TEST_DIR }}
+          docker cp ${{ env.IROHA_CONTAINER }}:/usr/local/bin/${{ env.IROHA_BIN }} ${{ env.TEST_DIR }}
+          cp ./defaults/client.toml ${{ env.TEST_DIR }}
+      - name: Make Binaries Executable
         run: |
-          poetry run pytest
-      - name: Install client cli dependencies using Poetry
+          chmod +x ${{ env.TEST_DIR }}/${{ env.IROHA_BIN }}
+      - name: Install Client CLI Dependencies
         working-directory: client_cli/pytests
         run: |
-          poetry install
-      - name: Run client cli tests
+          ${{ env.POETRY_PATH }} install
+      - name: Run Client CLI Tests
         working-directory: client_cli/pytests
         env:
-          # prepared by `test_env.py`
-          CLIENT_CLI_BINARY: ../../test/iroha
-          CLIENT_CLI_CONFIG: ../../test/client.toml
+          CLIENT_CLI_BINARY: ../../${{ env.TEST_DIR }}/${{ env.IROHA_BIN }}
+          CLIENT_CLI_CONFIG: ../../${{ env.TEST_DIR }}/client.toml
+        run: |
+          ${{ env.POETRY_PATH }} run pytest
+      - name: Tear Down Docker Compose Setup
+        run: docker compose -f ${{ env.DOCKER_COMPOSE_PATH }}/docker-compose.yml down
+
+  test-docker-compose-single-local:
```

Comment:
Move it back to the `torii-api-client-cli-docker-compose-tests` job. Doesn't make sense to increase paid runners time usage